### PR TITLE
GQSS-3732: added config validators

### PR DESCRIPTION
**Pull request by HrWiggles**
Recreated PR #97 against development branch

### CODE REVIEW

File: server/src/com/cloud/configuration/ConfigurationManagerImpl.java [hunk 1]
```diff
@@ -705,192 +703,46 @@ private String validateConfigurationValue(final String name, String value, final
...
-                }
-                if ("remote.access.vpn.psk.length".equalsIgnoreCase(name)) {
-                    if (val < 8) {
-                        throw new InvalidParameterValueException("Please enter a value greater than 8 for the configuration parameter:" + name);
-                    }
-                    if (val > 256) {
-                        throw new InvalidParameterValueException("Please enter a value less than 256 for the configuration parameter:" + name);
-                    }
+                if ("vm.password.length".equalsIgnoreCase(name)) {
+                    range = IntegerConfigValueRangeImpl.getRangeSingleton(10);
```
Comment (nnesic) [inline server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: Should probably lower this.

File: server/src/com/cloud/configuration/HypervisorListConfigValueRangeImpl.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @Override
+    public String validate(String name, String value) {
+        final String[] hypervisors = value.split(",");
+        if (hypervisors == null) {
+            return "Please enter hypervisor list, separated by comma";
+        } else {
+            for (final String hypervisor : hypervisors) {
+                if (HypervisorType.getType(hypervisor) == HypervisorType.Any || HypervisorType.getType(hypervisor) == HypervisorType.None) {
```
Comment (nnesic) [inline server/src/com/cloud/configuration/HypervisorListConfigValueRangeImpl.java]: Line too long

File: server/src/com/cloud/configuration/ConfigurationManagerImpl.java (hunk 1 above)
Comment (HrWiggles) [inline server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: Was explicitly increased to 10 in commit f53188192fb51fe8f7bb56bf71b43fc5c283f2ba
Even if we were to decide that we wanted to lower it, that should be a separate issue.

File: server/src/com/cloud/configuration/PredefinedListConfigValueRangeImpl.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+
+/**
+ * @author fridvin
+ */
+public class PredefinedListConfigValueRangeImpl implements PredefinedListConfigValueRange {
+    private static final Logger s_logger = Logger.getLogger(IntegerConfigValueRangeImpl.class);
```
Comment (nnesic) [inline server/src/com/cloud/configuration/PredefinedListConfigValueRangeImpl.java]: Shoudn't the logger be for PredefinedListConfigValueRangeImpl?

**Comment by nnesic** [comment]
Went over this quickly, as I had reviewed the previous pr before it was rebased. Looks good, just the few minor things.

File: server/src/com/cloud/configuration/ConfigurationManagerImpl.java (hunk 1 above)
Comment (nnesic) [inline server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: That's fair.

**Pull request merged**